src/modules/universal-discovery/components/content-meta-preview/content.meta.preview.component.js

```diff
@@ -159,6 +195,70 @@ export default class ContentMetaPreviewComponent extends Component {
         return <img className="c-meta-preview__image" src={this.state.imageUri} alt="" />;
     }
 
+    /**
+     * Removes bookmark
+     *
+     * @method removeBookmark
+     * @memberof ContentMetaPreviewComponent
+     */
+    removeBookmark() {
+        const { onBookmarkRemoved, data, restInfo } = this.props;
+        const locationId = data.id;
+        const unbookmarked = new Promise((resolve) => removeBookmark(restInfo, locationId, resolve));
+
+        unbookmarked
+            .then(() => {
+                this.setBookmarked(false);
+                onBookmarkRemoved(data);
+            })
+            .catch(showErrorNotification);
+    }
+
+    /**
+     * Adds bookmark
+     *
+     * @method addBookmark
+     * @memberof ContentMetaPreviewComponent
+     */
+    addBookmark() {
```

Comment:
`addBookmark` and `removeBookmark` are almost identical, can we combine this in one `toggleBookmark`?

Reply:
Good idea.